src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -2130,6 +2135,10 @@ private function registerMessengerConfiguration(array $config, ContainerBuilder
                 ['id' => 'handle_message'],
             ],
         ];
+        if (class_exists(LockMiddleware::class)) {
```

Comment:
we could also check for the existence of class LockFactory :thinking: 
If the lock component is not installed, let's avoid this by adding an extra middleware.

And LockFactory could be required in constructor's arguments

Reply:
Indeed, done